src/execution.rs

```diff
@@ -460,6 +598,20 @@ pub fn execute(
                         observed_task.stderr_pipe = None;
                     }
                 }
+                InputSource::Report => {
+                    if event.is_readable() {
+                        if let Some(ref mut pipe) = observed_task.report_pipe {
+                            let n = pipe.read(&mut buf).expect("failed to read REPORT");
+                            observed_task.report_decoder.append(&buf[0..n]);
+                            if let Some(stage_rep) = observed_task.report_decoder.try_decode() {
```

Comment:
```suggestion
                            while let Some(stage_rep) = observed_task.report_decoder.try_decode() {
```
With some buffering it's possible that there are more than one message in a single buffer.